org.adempiere.base/src/org/compiere/acct/Doc_MatchInv.java

```diff
@@ -339,20 +345,17 @@ public BigDecimal getBalance()
 				return null;
 		}
 		
-		if (m_matchInv.getReversal_ID() == 0) 
+		cr.setC_Activity_ID(m_invoiceLine.getC_Activity_ID());
+		cr.setC_Campaign_ID(m_invoiceLine.getC_Campaign_ID());
+		cr.setC_Project_ID(m_invoiceLine.getC_Project_ID());
+		cr.setC_ProjectPhase_ID(m_invoiceLine.getC_ProjectPhase_ID());
+		cr.setC_ProjectTask_ID(m_invoiceLine.getC_ProjectTask_ID());
+		cr.setC_UOM_ID(m_invoiceLine.getC_UOM_ID());
+		cr.setUser1_ID(m_invoiceLine.getUser1_ID());
+		cr.setUser2_ID(m_invoiceLine.getUser2_ID());
+		if (m_matchInv.getReversal_ID() > 0)

```

Comment:
can you check whether m_matchInv.isReversal() works here ? If yes, that's the prefer approach.

Reply:
Yes, using m_matchInv.isReversal() works here. 
There are many occurences of "if (m_matchInv.getReversal_ID() == 0) " and "if (m_matchInv.getReversal_ID() > 0)" in this file. Did you mean for all of them to be changed?

Reply:
Yes, I think we should change those as well.

Reply:
OK, will do the changes and test over the weekend.

Reply:
I have made the requested changes and all unit tests are passing.

Reply:
@hengsin,  After further testing it appears that the new method MMatchInv.isReversal() is not working as expected. With a Credit Memo the match qty = -1 and the related m_inout movement qty = +1, so when you reverse the credit memo the original match invoice returns true for isReversal(), not the newly created match invoice. Below is a possible solution, which is similar to how the existing code in Doc_MatchInv determines the true reversal in several places.

```
	public boolean isReversal() {
		if (getReversal_ID() > 0) {
			MMatchInv reversal = new MMatchInv (getCtx(), getReversal_ID(), get_TrxName());
			if (reversal.getM_MatchInv_ID() < getM_MatchInv_ID())
				return true;
		}
		return false;
	}
```
Using this method all existing tests still pass, plus a new test reversing a credit memo.

Reply:
hi @tony-tspc , yes, that looks more reliable. please add that in as well, thanks.

Reply:
hi @hengsin , all done and added a new test for isReversal() when reversing a credit memo.
Thanks, Tony